catalog/app/utils/usePrevious.js

```diff
@@ -1,5 +1,9 @@
 import * as React from 'react'
 
+/**
+ * Fire onChange callback with previous value and memoize current value
+ * TODO: explain why `useEffect`
```

Comment:
onChange callback is fired in useEffect so that one can execute effectful code there. for sync usage it should be just `const prev = usePrevious(...)`.
tho in #1973 i've fixed `usePrevious` to update the stored value synchronously based on this use case (when useEffect doesnt get called due to re-renders caused by setState)